pkg/jobworker/tail_test.go

```diff
@@ -0,0 +1,55 @@
+package jobworker
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"testing"
+	"time"
+)
+
+const testFile = "/tmp/tail_reader_test"
+const testLogs = "hello world\nhello test\n"
+const expectedLogs = "hello world\nhello test\ntesting 0\ntesting 1\ntesting 2\n"
+
+func Test_tailReader(t *testing.T) {
+	// Create a test file with known contents
+	err := os.WriteFile(testFile, []byte(testLogs), 0644)
+	if err != nil {
+		t.Error(err)
+	}
+	// Create tail reader and read from checking contents
+	reader, err := newTailReader(testFile, 20*time.Millisecond, true)
+	// Append some "testing n" to emulate a job logging to STDOUT
+	go func(r io.ReadCloser) {
+		// Close the tail reader once writing is done
+		defer r.Close()
+		// Open TEST file in append only
+		f, err := os.OpenFile(testFile, os.O_APPEND|os.O_WRONLY, 0644)
+		if err != nil {
+			panic(err)
+		}
+		defer f.Close()
+		// Append some logging output
+		for i := range 3 {
+			if _, err = f.WriteString(fmt.Sprintf("testing %d\n", i)); err != nil {
+				t.Error("failed to write string to test file", err)
+				return
+			}
+			time.Sleep(50 * time.Millisecond)
+		}
+	}(reader)
+	// Read the job's "output" written from the goroutine above
+	scanner := bufio.NewScanner(reader)
+	recievedLogs := ""
+	for scanner.Scan() {
```

Comment:
In the real world I'd probably want some sort of short timeout on this, so a deadlocked test doesn't gum up CI until the overall test timeout expires. 

A quick `TODO` about how you'd approach this will do for now.